test/e2e/data/infrastructure-docker/main/clusterclass-quick-start-runtimesdk-generateplan.yaml

```diff
@@ -0,0 +1,147 @@
+apiVersion: cluster.x-k8s.io/v1beta2
+kind: ClusterClass
+metadata:
+  name: quick-start-runtimesdk-generateplan
+spec:
+  controlPlane:
+    templateRef:
+      apiVersion: controlplane.cluster.x-k8s.io/v1beta2
+      kind: KubeadmControlPlaneTemplate
+      name: quick-start-control-plane
+    machineInfrastructure:
+      templateRef:
+        apiVersion: infrastructure.cluster.x-k8s.io/v1beta2
+        kind: DockerMachineTemplate
+        name: quick-start-control-plane
+    naming:
+      template: "{{ .cluster.name }}-cp-{{ .random }}"
+  infrastructure:
+    templateRef:
+      apiVersion: infrastructure.cluster.x-k8s.io/v1beta2
+      kind: DockerClusterTemplate
+      name: quick-start-cluster
+    naming:
+      template: "{{ .cluster.name }}-infra-{{ .random }}"
+  workers:
+    machineDeployments:
+    - class: default-worker
+      naming:
+        template: "{{ .cluster.name }}-md-{{ .machineDeployment.topologyName }}-{{ .random }}"
+      bootstrap:
+        templateRef:
+          apiVersion: bootstrap.cluster.x-k8s.io/v1beta2
+          kind: KubeadmConfigTemplate
+          name: quick-start-default-worker-bootstraptemplate
+      infrastructure:
+        templateRef:
+          apiVersion: infrastructure.cluster.x-k8s.io/v1beta2
+          kind: DockerMachineTemplate
+          name: quick-start-default-worker-machinetemplate
+    machinePools:
+    - class: default-worker
+      naming:
+        template: "{{ .cluster.name }}-mp-{{ .machinePool.topologyName }}-{{ .random }}"
+      bootstrap:
+        templateRef:
+          apiVersion: bootstrap.cluster.x-k8s.io/v1beta2
+          kind: KubeadmConfigTemplate
+          name: quick-start-default-worker-bootstraptemplate
+      infrastructure:
+        templateRef:
+          apiVersion: infrastructure.cluster.x-k8s.io/v1beta2
+          kind: DockerMachinePoolTemplate
+          name: quick-start-default-worker-machinepooltemplate
+  patches:
+  - name: test-patch
+    external:
+      generatePatchesExtension: generate-patches.${EXTENSION_CONFIG_NAME:-k8s-upgrade-with-runtimesdk}
+      validateTopologyExtension: validate-topology.${EXTENSION_CONFIG_NAME:-k8s-upgrade-with-runtimesdk}
+      discoverVariablesExtension: discover-variables.${EXTENSION_CONFIG_NAME:-k8s-upgrade-with-runtimesdk}
+  upgrade:
+    external:
+      generateUpgradePlanExtension: generate-upgrade-plan.${EXTENSION_CONFIG_NAME:-k8s-upgrade-with-runtimesdk}
```

Comment:
@sivchari if I got this right, this is the only difference between the test introduced by this PR and the existing runtime SDK test.

If this is correct, considering that this test takes long, I would be ok if we move this change to clusterclass-quick-start-runtimesdk and have only one test (I'm also considering that we have good test coverage for the different ways of computing upgrade plan, and that the runtime extension internally uses the same func we  use when only kubernetes versions are set).

wdyt?
@sbueringer opinions?

Reply:
Sounds good. Let's just add this to the clusterclass-quick-start-runtimesdk ClusterClass and revert the rest of the changes on this PR

@sivchari Sorry we didn't think about this when writing up the tasks

Reply:
Thanks!, I addressed it.